tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/OpenTelemetry/ResourceAttributeProcessorHelper.cs

```diff
@@ -0,0 +1,96 @@
+// <copyright file="ResourceAttributeProcessorHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Reflection.Emit;
+using Datadog.Trace.Activity.DuckTypes;
+using Datadog.Trace.Activity.Handlers;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.OpenTelemetry
+{
+    internal static class ResourceAttributeProcessorHelper
+    {
+        private static Func<object, object>? _getResourceDelegate;
+
+        static ResourceAttributeProcessorHelper()
+        {
+            _getResourceDelegate = CreateGetResourceDelegate();
+        }
+
+        public static void OnStart(object processor, object activityData)
+        {
+            if (_getResourceDelegate is null
+                || !activityData.TryDuckCast<IActivity>(out var activity)
+                || !processor.TryDuckCast<BaseProcessorStruct>(out var baseProcessor))
+            {
+                return;
+            }
+
+            if (DefaultActivityHandler.ActivityMappingById.TryGetValue(activity.Id, out var activityMapping))
+            {
+                if (baseProcessor.ParentProvider is not null)
+                {
+                    var resourceObject = _getResourceDelegate(baseProcessor.ParentProvider);
+                    if (resourceObject.TryDuckCast<IResource>(out var resource))
+                    {
+                        var span = activityMapping.Scope.Span;
+                        foreach (var attribute in resource.Attributes)
+                        {
+                            // In addition to copying the attribute as a tag, update span fields for specific keys
+                            if (attribute.Key == "service.name")
+                            {
+                                span.ServiceName = attribute.Value.ToString();
+                            }
+                            else if (attribute.Key == "service.version")
+                            {
+                                span.SetTag(Tags.Version, attribute.Value.ToString());
+                            }
+
+                            span.SetTag(attribute.Key, attribute.Value.ToString());
```

Comment:
I think we want this in an `else`?
```suggestion
                            else
                            {
                                span.SetTag(attribute.Key, attribute.Value.ToString());
                            }
```

Reply:
In this case I intentionally always want to set the tag as well so I plan to revert this particular change. The `if / else if` above is supposed to be additive, I can try to make that more clear in the code comments

Reply:
Implemented my above comment in 9dbfde3e8eebc2ed9023694b7dad341b04de889c